packages/rest/test/integration/http-handler.integration.ts

```diff
@@ -418,6 +431,37 @@ describe('HttpHandler', () => {
       logErrorsExcept(500);
 
       await client.get('/hello').expect(500, {
+        error: {
+          message: 'Internal Server Error',
+          statusCode: 500,
+        },
+      });
+    });
+
+    it('respects error handler options', async () => {
+      rootContext.bind(RestBindings.ERROR_WRITER_OPTIONS).to({debug: true});
```

Comment:
a question: how is the option `{debug: true}` tested?

Reply:
When the error handler is set to `debug` mode, it returns details of 500 errors in the HTTP responses. By default, `debug` is disabled and HTTP responses show only a generic HTTP status message.

This is important for security to prevent the error handler from leaking sensitive informations like filesystem paths or hostnames & ports. This information is often included in the error message and/or error properties.

```
$ node
> fs.readFileSync('/etc/passwords')
Error: ENOENT: no such file or directory, open '/etc/passwords'
```

See also the assertion on lines L434-438 above, where the error handler in production mode returns the following output:

```
        error: {
           message: 'Internal Server Error',
           statusCode: 500,
         },
```

Compare it with debug mode:

```
        error: {
           message: 'Bad hello',
           statusCode: 500,
         },
```

Reply:
@bajtos 👍 Thanks for the detailed answer, got it. Flag `debug` controls the security level which affects the returned error message.